web/modules/custom/dpl_event/dpl_event.module

```diff
@@ -186,17 +187,23 @@ function dpl_event_preprocess_eventseries(array &$variables): void {
     return;
   }
 
-  // Load event instances related to this series and prepare them for rendering.
+  // Load event instances & filter out past events.
   $eventinstances_in_series = $event_series->get('event_instances')->referencedEntities();
+  $current_date = (new DateTime())->setTime(0, 0, 0);
 
-  $variables['event_instances'] = [];
-  foreach ($eventinstances_in_series as $index => $eventInstance) {
-    $viewMode = $index === 0 ? 'list_teaser_stacked_parent' : 'stacked_event';
+  $eventinstances_in_series_filtered = array_filter($eventinstances_in_series, function ($eventInstance) use ($current_date) {
+    $event_start_date = new DateTime($eventInstance->get('date')->start_date);
+    return $event_start_date >= $current_date;
+  });
 
-    $renderable_instances = \Drupal::entityTypeManager()
-      ->getViewBuilder('eventinstance')->view($eventInstance, $viewMode);
+  $eventinstances_in_series_filtered = array_values($eventinstances_in_series_filtered);
 
-    $variables['event_instances'][] = $renderable_instances;
+  $viewBuilder = \Drupal::entityTypeManager()->getViewBuilder('eventinstance');
+  $variables['event_instances'] = [];
+  foreach ($eventinstances_in_series_filtered as $index => $eventInstance) {
+    $viewMode = $index === 0 ? 'list_teaser_stacked_parent' : 'stacked_event';
+    $renderable_instance = $viewBuilder->view($eventInstance, $viewMode);
+    $variables['event_instances'][] = $renderable_instance;
```

Comment:
```
$variables['#cache'] =  [
        // Max age of 12 hours.
        '#max-age' => (60 * 60 * 12),
        '#tags' => ['eventinstance_list', 'eventseries_list'],
]

Reply:
You could also set the current date as a cache context.